source/fundamentals/crud/bulk.txt

```diff
@@ -0,0 +1,315 @@
+   :end-before: end insertDocumentsExample
+
+.. note::
+
+   When performing a ``bulkWrite()``, the ``InsertOneModel`` cannot
+   insert a document with an ``_id`` that already exists in the
+   collection. Instead, the method throws an ``MongoBulkWriteException``.
+
+   The following example tries to insert two documents where the ``_id`` is
+   "1" and "3":
+   
+   .. literalinclude:: /includes/fundamentals/code-snippets/BulkWrite.java
+      :language: java
+      :dedent:
+      :start-after: begin insertExceptionExample
+      :end-before: end insertExceptionExample
+
+   The following shows the output of the code above:
+
+   .. code-block::
+
+      Documents in the collection: 
+      {"_id": 1}
+      {"_id": 2}
+
+      A MongoBulkWriteException occured: A MongoBulkWriteException occured with the following message:  
+      Bulk write operation error on server driver-ref-shard-00-02.pw0q4.mongodb.net:27017. 
+      Write errors: [BulkWriteError{index=0, code=11000, message='E11000 duplicate key error collection: 
+      crudOps.bulkWrite index: _id_ dup key: { _id: 1 }', details={}}].
+
+   To see why the document where the ``_id`` is "3" didn't insert, see
+   the :ref:`Order of Execution <orderOfExecution>` section.
+
+For additional information on the classes and methods mentioned in this
+section, see the following resources: 
+
+- :java-core-api:`InsertOneModel <com/mongodb/client/model/InsertOneModel.html>` API Documentation
+
+Replace Operation
+~~~~~~~~~~~~~~~~~
+
+To perform a replace operation, create an ``ReplaceOneModel`` specifying
+the document you want to replace with the replacement document. 
+
+.. note::
+
+   When performing a ``bulkWrite()``, the ``ReplaceOneModel`` cannot
+   make changes to a document that violate unique index constraints on
+   the collection. Also, the model does not replace a document if there
+   are no matches to your query filter. 
+
+Example
+```````
+
+The following example creates a ``ReplaceOneModel`` to
+replace a document where the ``_id`` is "1" to a document with an
+additional field ``x`` with the value "4": 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/BulkWrite.java
+   :language: java
+   :dedent:
+   :start-after: begin replaceDocumentsExample
+   :end-before: end replaceDocumentsExample
+
+For additional information on the classes and methods mentioned in this
+section, see the following resources: 
+
+- :java-core-api:`ReplaceOneModel <com/mongodb/client/model/ReplaceOneModel.html>` API Documentation
+- :manual:`Unique indexes </core/index-unique/>` Server Manual Explanation
+
+Update Operation
+~~~~~~~~~~~~~~~~
+
+To perform an update operation, create an ``UpdateOneModel`` or an
+``UpdateManyModel`` specifying the document you want to update with what
+the updates are.
+
+The ``UpdateOneModel`` updates one document that matches your query
+filter and the ``UpdateManyModel`` updates all the documents that
+match your query filter. 
```

Comment:
suggestion: clarify and expand. While correct it updates one document, does it update the `first` document that matches the query filter?